x-pack/plugins/security_solution/public/detections/components/user_privileges/index.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, useContext } from 'react';
+import { useFetchDetectionEnginePrivileges } from './use_fetch_detection_engine_privileges';
+import { useFetchListPrivileges } from './use_fetch_list_privileges';
+
+export interface UserPrivilegesState {
+  listPrivileges: ReturnType<typeof useFetchListPrivileges>;
+  detectionEnginePrivileges: ReturnType<typeof useFetchDetectionEnginePrivileges>;
+}
+
+const UserPrivilegesContext = createContext<UserPrivilegesState>({
+  listPrivileges: { loading: false, error: undefined, result: undefined },
+  detectionEnginePrivileges: { loading: false, error: undefined, result: undefined },
+});
+
+interface UserPrivilegesProviderProps {
+  children: React.ReactNode;
+}
+
+export const UserPrivilegesProvider = ({ children }: UserPrivilegesProviderProps) => {
```

Comment:
@yctercero @dhurley14, if I'm not mistaken, you are currently working on RBAC. Could you please take a look at these changes? Do they conflict with your current work? Maybe we should sync in that case.

Reply:
Hmm that's a good question. So we would no longer be checking the ES privileges but instead going through the Kibana feature/privileges model and the client currently being built out.

I believe a lot of these changes would still be incredibly useful, how we check for permission might just change as well as what instructions we give the user since they would no longer be adding ES permissions, but modifying the kibana user role privileges instead. I'd maybe pull in @spong and @peluja1012 here to figure out timelines and some overlap. I don't see why this shouldn't go in right now as it is an improvement on the existing experience and then Devin and I can work off of this to make any necessary updates once RBAC is ready?

Reply:
> I don't see why this shouldn't go in right now as it is an improvement on the existing experience and then Devin and I can work off of this to make any necessary updates once RBAC is ready?

++, that sounds good to me @yctercero. 👍 Even with RBAC, we'll still have use-cases for determining if the required ES privileges are present, like if we have an embeddable dashboard/viz on the page, or if we end up adding an option to TGrid to bypass Kibana RBAC in favor of ES privileges.